### PR TITLE
[MXNET-312] Added Matthew's Correlation Coefficient to metrics

### PR DESCRIPTION
**Pull request by dabraude**
## Description ##
Added Matthew's Correlation Coefficient to metrics

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [X] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Added MCC binary classification metric (and when applicable, API doc)

## Comments ##
- In documentation gave example where it shows the difference with F1

### CODE REVIEW

**Comment by dabraude** [comment]
@marcoabreu can you check the CI?

**Comment by marcoabreu** [comment]
What's the matter?

**Comment by dabraude** [comment]
I meant can you check the test I added is ok

File: python/mxnet/metric.py [hunk 1]
```diff
@@ -661,6 +682,107 @@ def reset(self):
...
+    >>> print f1.get()
+    ('f1', 0.95233560306652054)
+    >>> print mcc.get()
+    ('mcc', 0.01917751877733392)
+    """
+
+    def __init__(self, name='mcc',
+                 output_names=None, label_names=None, average="macro"):
+        self.average = average
+        self.metrics = _BinaryClassificationMetrics()
```
Comment (piiswrong) [inline python/mxnet/metric.py]: _average and _metrics

File: python/mxnet/metric.py [hunk 2]
```diff
@@ -661,6 +682,107 @@ def reset(self):
...
 
 
+@register
+class MCC(EvalMetric):
+    """Computes the Matthews Correlation Coefficient of a binary classification problem.
+
+    While slower to compute the MCC can give insight that F1 or Accuracy cannot.
+    For instance, if the network always predicts the same result
+    then the MCC will immeadiately show this. The MCC is also symetric with respect
+    to positive and negative catagorisation, however, there needs to be both
```
Comment (ThomasDelteil) [inline python/mxnet/metric.py]: categorisation*

**Comment by dabraude** [comment]
@piiswrong  Would you mind updating your review?

**Comment by dabraude** [comment]
Hey is there anything I can fix on this?

File: tests/python/unittest/test_metric.py [hunk 3]
```diff
@@ -122,6 +123,54 @@ def test_f1():
     np.testing.assert_almost_equal(microF1.get()[1], fscore_total)
     np.testing.assert_almost_equal(macroF1.get()[1], (fscore1 + fscore2) / 2.)
 
+def test_mcc():
```
Comment (szha) [inline tests/python/unittest/test_metric.py]: could you add a consistency test with Matthew's Correlation Coefficient in numpy?

Comment (szha) [reply tests/python/unittest/test_metric.py]: never mind. I didn't realize that numpy doesn't have it.

**Pull request merged**